File: backend/app/api/recommend.py
from fastapi import APIRouter

from app.schemas.recommend import (
    RecommendRequest,
    RecommendResponse,
    VariantRequest,
    VariantResponse,
)
from app.services.question_generation_service import generate_variant_questions
from app.services.recommend_service import recommend_similar_questions


router = APIRouter(tags=["recommend"])


@router.post("/recommend", response_model=RecommendResponse)
def recommend(req: RecommendRequest):
    return RecommendResponse(
        questions=recommend_similar_questions(
            knowledge_points=req.knowledge_points,
            difficulty=req.difficulty,
            count=req.count,
        )
    )


@router.post("/recommend/variants", response_model=VariantResponse)
def recommend_variants(req: VariantRequest):
    return VariantResponse(
        questions=generate_variant_questions(
            question_text=req.question_text,
            knowledge_points=req.knowledge_points,
            difficulty=req.difficulty,
            count=req.count,
        )
    )

File: frontend/src/api/recommend.ts
import { http } from './http'

const RECOMMEND_HTTP_TIMEOUT_MS = 120_000

export interface GeneratedQuestion {
  question_text: string
  answer: string
  solution: string
  difficulty: string
  knowledge_points: string[]
  question_type: string
  source: string
}

export interface VariantResponse {
  questions: GeneratedQuestion[]
}

export function generateVariantQuestions(payload: {
  question_text: string
  knowledge_points: string[]
  difficulty?: string
  count?: number
}) {
  return http.post<VariantResponse>('/recommend/variants', payload, {
    timeout: RECOMMEND_HTTP_TIMEOUT_MS,
  })
}
